front/app/components/admin/GraphCards/CommentsByTimeCard/useCommentsByTime/parse.ts

```diff
@@ -0,0 +1,123 @@
+import moment, { Moment } from 'moment';
+
+// utils
+import { timeSeriesParser } from 'components/admin/GraphCards/_utils/timeSeries';
+import { get } from 'utils/helperUtils';
+import { orderBy } from 'lodash-es';
+
+// typings
+import {
+  Response,
+  TimeSeriesResponseRow,
+  TimeSeries,
+  TimeSeriesRow,
+} from './typings';
+import { Translations } from './translations';
+import { IResolution } from 'components/admin/ResolutionControl';
+
+export const getEmptyRow = (date: Moment) => ({
+  date: date.format('YYYY-MM-DD'),
+  comments: 0,
+  total: 0,
+});
+
+const parseRow = (date: Moment, row?: TimeSeriesResponseRow): TimeSeriesRow => {
+  if (!row) return getEmptyRow(date);
+
+  return {
+    total: 0,
+    comments: row.count,
+    date: date.format('YYYY-MM-DD'),
+  };
+};
+
+const getDate = (row: TimeSeriesResponseRow) => {
+  return moment(get(row, 'first_dimension_date_created_date'));
+};
+
+const _parseTimeSeries = timeSeriesParser(getDate, parseRow);
+
+export const parseTimeSeries = (
+  responseTimeSeries: Response['data'][0],
+  startAtMoment: Moment | null | undefined,
+  endAtMoment: Moment | null,
+  resolution: IResolution,
+  total: Response['data'][1]
+): TimeSeries | null => {
+  let timeSeries = _parseTimeSeries(
+    responseTimeSeries,
+    startAtMoment,
+    endAtMoment,
+    resolution
+  );
+  if (!timeSeries) return [];
+  let totalCount = total && total.length > 0 ? total[0]?.count : 0;
```

Comment:
Why are we requesting the total from the API, but is it possible that's it's undefined, in which case we set it to zero? I think if we can't rely on the total coming back from the API to be correct, we shouldn't be using it at all, and just derive it from the time series instead

Reply:
yeah we can assume to be present if it got to that point, so removing that check.
but the series brings data from that period only so is necesary to always get the global total from another query.